process/process_posix.go

```diff
@@ -69,6 +70,34 @@ func getTerminalMap() (map[uint64]string, error) {
 	return ret, nil
 }
 
+func PidExistsWithContext(ctx context.Context, pid int32) (bool, error) {
+	if pid <= 0 {
+		return false, fmt.Errorf("invalid pid %v", pid)
+	}
+	proc, err := os.FindProcess(int(pid))
+	if err != nil {
+		return false, err
+	}
+	err = proc.Signal(syscall.Signal(0))
```

Comment:
This line actually breaks the whole story if you are running inside container, and you are tracking host processes.
More details can be found here:
https://github.com/influxdata/telegraf/issues/6813